src/Catalog/Helpers/FeedHelpers.cs

```diff
@@ -268,76 +274,23 @@ private static DateTime ForceUtc(DateTime date)
                 return lastDate;
             }
 
-            foreach (var entry in packages)
+            // Flatten the sorted list.
+            var workItems = packages.SelectMany(
+                    pair => pair.Value.Select(
+                        details => new PackageWorkItem(pair.Key, details)))
+                .ToArray();
```

Comment:
Unnecessary `ToArray()`? `ForEachAsync` call below should work fine with any `IEnumerable`.

Reply:
Not unnecessary.  The enumerable is used [below](https://github.com/NuGet/NuGet.Services.Metadata/pull/323/files/1e39f6ca9a5a2dd713f8bca15fb7415fb199b2a7#diff-ef9bbac124ec39795cf902ad6b03bfc2R291) again.